File: solutions/javascript/grains/1/grains.js
export const square = (squareNumber, returnTotalGrains) => {
  if (squareNumber < 1 || squareNumber > 64) {
    throw 'square must be between 1 and 64';
  }

  const grains = {
    square: BigInt(1),
    total: BigInt(1)
  }

  for (let i = 0; i < (squareNumber - 1); i++) {
    grains.square += grains.square;
    grains.total  += grains.square;
  }

  return returnTotalGrains ? grains.total : grains.square;
};

export const total = () => {
  const returnTotalGrains = true;
  return square(64, returnTotalGrains);
};

File: solutions/python/grains/1/grains.py
def square(number):
    if number < 1 or number > 64:
        raise ValueError("square must be between 1 and 64")
    
    return pow(2, number - 1)

    

def total():
    total_grains_on_the_board = 0

    for i in range(64):
        total_grains_on_the_board += square(i + 1)

    return total_grains_on_the_board